code/iaas/lb-instance/src/main/java/io/cattle/platform/lb/instance/process/LoadBalancerUpdateConfig.java

```diff
@@ -208,8 +212,13 @@ private void updatePublicPorts(ProcessState state, Map<Long, List<? extends Inst
             List<? extends Instance> lbInstances = lbInstancesMap.get(lbId);
             // surround by lock
             for (final Instance lbInstance : lbInstances) {
-                final boolean hasActiveInstances = (targetDao.getLoadBalancerActiveInstanceTargets(lb.getId()).size() + targetDao
+                boolean lbHasActiveTargets = (targetDao.getLoadBalancerActiveInstanceTargets(lb.getId()).size() + targetDao
                         .getLoadBalancerActiveIpTargets(lb.getId()).size()) > 0;
+                List<String> ports = DataUtils.getFieldList(lbInstance.getData(), InstanceConstants.FIELD_PORTS,
+                        String.class);
+                boolean lbInstanceHasPortsSet = (ports != null && !ports.isEmpty());
+                final boolean hasActiveInstances = lbHasActiveTargets || lbInstanceHasPortsSet;
```

Comment:
Before this refactoring, lb ports binding to host were dynamic; and we were opening/closing them only when LoadBalancer had active (Running) targets. Now to make it work with the new allocator, initial set of listener ports is set on the lb system instance itself, and we have to open the ports in this case on the host even when there are no active targets exist in the load balancer. Therefore to determine if the ports have to be opened, we check 2 conditions:

* lbHasActiveTargets
* lbInstanceHasPortsSet

If either one of them true, we open the ports